File: python/94 PriQ.py
import heapq
class Student:
    def __init__(self,name,grade):
        self.name=name
        self.grade=grade

    def __lt__(self, other):
        return self.grade < other.grade
    
    def __str__(self):
        return f"{self.name}({self.grade})"
pq=[]
heapq.heappush(pq, Student("Alice", 90))
heapq.heappush(pq, Student("Bob", 75))
heapq.heappush(pq, Student("Charlie", 85))
heapq.heappush(pq, Student("Dave", 95))

print("Priority Queue (by grade):")
while pq:
    print(heapq.heappop(pq))

File: java/94 PriQ.java
package code;
import java.util.Comparator;
import java.util.PriorityQueue;
import java.util.Queue;
class Student{
	String name;
	int grade;
	public Student(String name, int grade) {
		this.name = name;
		this.grade = grade;
	}
	
}
public class Test {
	public static void main(String[] args) {
		Queue<Student> set1=new PriorityQueue<>(new Comparator<Student>() {

			@Override
			public int compare(Student o1, Student o2) {
				if (o1.grade==o2.grade) {
					return 0;
				}
				else if(o1.grade>o2.grade) {
					return -1;
				}
				else {
					return 1;
				}
				
			}
		});
		Student s1=new Student("Abc",90);
		Student s2=new Student("ABc",80);
		Student s3=new Student("ABC",85);
		set1.add(s1);
		set1.add(s2);
		set1.add(s3);
		System.out.println(set1);
	}
}